specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2022-02-15-preview/cosmos-db.json

```diff
@@ -8263,6 +8507,198 @@
         }
       }
     },
+    "RedistributeThroughputParameters": {
+      "description": "Cosmos DB redistribute throughput parameters object",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Properties to redistribute throughput parameters object",
+          "$ref": "#/definitions/RedistributeThroughputProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/ARMResourceProperties"
+        }
+      ],
+      "required": [
+        "properties"
+      ]
+    },
+    "RedistributeThroughputProperties": {
+      "description": "Properties to redistribute throughput for Azure Cosmos DB resource.",
+      "type": "object",
+      "properties": {
+        "resource": {
```

Comment:
>resource [](http://example.com/codeflow?start=9&length=8)

What is this wrapper for?

Reply:
we do this so we can introduce "options" like done elsewhere in the spec, for example here : https://github.com/Azure/azure-rest-api-specs/blob/b5d8eef5bf187dc99f1c18e8323b19e0c37a7124/specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2022-02-15-preview/cosmos-db.json#L6643-L6660